File: mini-program/src/api/video.ts
// 视频相关接口
import { http } from './request'

export interface VideoItem {
  id: number
  title: string
  description: string
  video_url: string
  cover_url: string
  duration: number // 秒
  view_count: number
  sort_order: number
  created_at: string
}

export interface VideoViewResult {
  video_id: number
  view_count: number
  viewed: boolean
}

export function getVideos() {
  return http.get<VideoItem[]>('/videos')
}

export function getVideo(id: number) {
  return http.get<VideoItem>(`/videos/${id}`)
}

export function reportVideoView(video_id: number, viewer_key: string) {
  return http.post<VideoViewResult>(
    `/videos/${video_id}/view`,
    { viewer_key },
    false, // 游客也可观看,不带 token 也行(带也行)
  )
}

File: server/app/schemas/video.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class VideoOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    description: str = ""
    video_url: str
    cover_url: str = ""
    duration: int = 0
    view_count: int = 0
    sort_order: int = 0
    created_at: datetime


class VideoViewOut(BaseModel):
    video_id: int
    view_count: int
    viewed: bool  # 本次是否算新增浏览量